pkg/resources/reconciling/wrapper.go

```diff
@@ -142,6 +143,22 @@ func DefaultPodSpec(oldPodSpec, newPodSpec corev1.PodSpec) (corev1.PodSpec, erro
 		}
 	}
 
+	// set KKP specific defaults for every Pod created by it
+
+	if newPodSpec.SecurityContext == nil {
+		newPodSpec.SecurityContext = &corev1.PodSecurityContext{
+			SeccompProfile: &corev1.SeccompProfile{
+				Type: corev1.SeccompProfileTypeRuntimeDefault,
+			},
+		}
+	}
+
+	if newPodSpec.SecurityContext != nil && newPodSpec.SecurityContext.SeccompProfile == nil {
```

Comment:
```suggestion
	if newPodSpec.SecurityContext.SeccompProfile == nil {
```

checking here `newPodSpec.SecurityContext` for nilness is useless since you just initialized it above

Reply:
Good point. I initially misunderstood the function and was checking `oldPodSpec`, but now it does not make sense anymore to check for `nil`.